### PR TITLE
Add search for virtual disk image

### PR DESCRIPTION
**Pull request by mbhavya**
This PR adds search with id functionality for VDIs (similar to search_with_id in VirtualServer).

Hi @SLsthompson , 
Could you please take a look and let me know if this PR needs more work before merging ? 
Thanks
Bhavya ( @mbhavya )

### CODE REVIEW

**Comment by renier** [comment]
@SLsthompson FYA, the checks in this pull request fails only when testing on JRuby19, because of the coveralls->tins dependency (_tins_ requires Ruby 2.0).

So, should we remove jruby19 testing from .travis.yml? Or should we use something other than coveralls for the coverage testing? e.g. simplecov

File: lib/softlayer/VirtualDiskImage.rb [hunk 1]
```diff
@@ -188,6 +188,35 @@ def service
...
+    #
+    # The options parameter should contain:
+    #
+    # <b>+:client+</b> - The client used to connect to the API
+    #
+    # If no client is given, then the routine will try to use Client.default_client
+    # If no client can be found the routine will raise an error.
+    #
+    # The options may include the following keys
+    # * <b>+:object_mask+</b> (string) - A object mask of properties, in addition to the default properties, that you wish to retrieve for the server
```
Comment (renier) [inline lib/softlayer/VirtualDiskImage.rb]: Should say "for the image", right?

Comment (renier) [inline lib/softlayer/VirtualDiskImage.rb]: Should say "for the image", right?

Comment (mbhavya) [inline lib/softlayer/VirtualDiskImage.rb]: yes!
Give me a few mins, correcting it now...

**Pull request merged**